src/app/cli/src/commands/ui_command.rs

```diff
@@ -61,49 +63,56 @@ impl UICommand {
 #[async_trait::async_trait(?Send)]
 impl Command for UICommand {
     async fn run(&mut self) -> Result<(), CLIError> {
-        let web_server = crate::explore::WebUIServer::new(
-            self.base_catalog.clone(),
-            self.multi_tenant_workspace,
-            self.current_account_name.clone(),
-            self.predefined_accounts_config.clone(),
-            self.address,
-            self.port,
-        )
-        .await;
-
-        let web_server_url = format!("http://{}", web_server.local_addr());
-        tracing::info!("HTTP server is listening on: {}", web_server_url);
-
-        if self.get_token {
-            tracing::info!(
-                token = %web_server.get_access_token(),
-                "Issued API server token"
-            );
-        }
+        // The borrow checker is happy if self is not explicitly used
+        let self_ref = &self;
+
+        run_transactional([&self.base_catalog], move |[base_catalog]| async move {
```

Comment:
How is this passing compilation? I don't see `run_transactional` accepting an array...

Reply:
Yes, this won't compile, and I guess, @s373r hasn't tried compiling in Web UI mode.

Reply:
It was working, until I didn't get https://github.com/kamu-data/kamu-cli/pull/650#discussion_r1612147580.

I certainly tried compiling, before requesting a review for you -- the issue here is different in that after requesting the changes, I started making them, and didn't mark the PR status to draft.

Right now PR is in WIP state, so some things may not compile

Reply:
Done (and double-checked)